node/src/Pos/Update/Download.hs

```diff
@@ -106,36 +110,43 @@ downloadUpdateDo cps@ConfirmedProposalState {..} = do
             throwError "There's unapplied update already downloaded"
 
         logInfo "Downloading update..."
-        file <- ExceptT $ liftIO (downloadHash updateServers updHash) <&>
+        file <- ExceptT $ downloadHash updateServers updHash <&>
                 first (sformat ("Update download (hash "%build%
                                 ") has failed: "%stext) updHash)
 
+        logInfo $ "Update was downloaded, saving to " <> show updPath
+
         liftIO $ BSL.writeFile updPath file
-        logInfo "Update was downloaded"
+        logInfo $ "Update was downloaded, saved to " <> show updPath
         downloadedMVar <- views (lensOf @UpdateContext) ucDownloadedUpdate
         putMVar downloadedMVar cps
         logInfo "Update MVar filled, wallet is notified"
 
-    whenLeft res logWarning
+    whenLeft res logDownloadError
   where
+    handleErr = return . Left . show
+    logDownloadError e =
+        logWarning $ sformat
+            ("Failed to download update proposal "%build%": "%stext)
+            cpsUpdateProposal e
     -- Check that we really should download an update with given
     -- 'SoftwareVersion'.
     isVersionAppropriate :: SoftwareVersion -> Bool
     isVersionAppropriate ver = svAppName ver == svAppName curSoftwareVersion
         && svNumber ver > svNumber curSoftwareVersion
 
-
 -- Download a file by its hash.
 --
 -- Tries all servers in turn, fails if none of them work.
-downloadHash :: [Text] -> Hash LByteString -> IO (Either Text LByteString)
+downloadHash :: UpdateMode ctx m => [Text] -> Hash LByteString -> m (Either Text LByteString)
```

Comment:
This constraint is too much for this function (it only needs `MonadIO` and `WithLogger`), but it's minor, so I don't care much.